src/main/java/backend/UpdateController.java

```diff
@@ -107,32 +111,37 @@ public void processAndRefresh(List<FilterPanel> filterPanels) {
     }
 
     /**
-     * Filters, sorts and counts issues within the model according to the given filter expressions.
+     * Filters, sorts and counts issues within the model according to the given panels' filter expressions.
      * In here, "processed" is equivalent to "filtered, sorted and counted".
      *
-     * @param filterExprs Filter expressions to process.
+     * @param filterPanels Filter panels to process.
      * @return Filter expressions and their corresponding issues after filtering, sorting and counting.
      */
-    private Map<FilterExpression, List<GuiElement>> processFilter(List<FilterExpression> filterExprs) {
+    private Map<FilterExpression, List<GuiElement>> processPanels(List<FilterPanel> filterPanels) {
         MultiModel models = logic.getModels();
         List<TurboIssue> allModelIssues = models.getIssues();
 
         Map<FilterExpression, List<GuiElement>> processed = new HashMap<>();
 
-        filterExprs.stream().distinct().forEach(filterExpr -> {
-            boolean hasUpdatedQualifier = Qualifier.hasUpdatedQualifier(filterExpr);
+        filterPanels.stream().distinct().forEach(filterPanel -> {
+            try {
+                FilterExpression filterExpr = filterPanel.getCurrentFilterExpression();
+                boolean hasUpdatedQualifier = Qualifier.hasUpdatedQualifier(filterExpr);
```

Comment:
Do these two lines need to be in the try block? We should try to minimise its scope.